plugin/adsk/scripts/AETemplateHelpers.py

```diff
@@ -35,4 +45,162 @@ def ProxyShapePayloads(proxyShape, loadAll):
                 cmd = contextOps.doOpCmd([cmdStr])
                 ufeCmd.execute(cmd)
     except:
+        debugMessage('ProxyShapePayloads() - error during function')
+        pass
+
+def IsProxyShapeLayerStackDirty(proxyStage):
+    # Helper method which returns True if any layer (skipping SessionLayer)
+    # in the layer stack is dirty.
+    if proxyStage:
+        # Is any layer in the layerstack dirty?
+        for layer in proxyStage.GetLayerStack(includeSessionLayers=False):
+            debugMessage('  Proessing layer: %s' % layer.GetDisplayName())
+            if layer.dirty:
+                debugMessage('    Found dirty layer')
+                return True
+    return False
+
+def GetStageFromProxyShapeAttr(attr):
+    # Helper method which returns the proxy shape name and Usd stage
+    # from the input attribute on the proxy shape.
+
+    # First get the stage name from the input attribute.
+    stageName = mel.eval('plugNode("%s")' % attr)
```

Comment:
Only thing plugNode.mel does is this:
tokenize($plug, ".", $buffer);
return $buffer[0];
Feels like we can do this pretty easily using Python string manipulation.

Reply:
Right, good idea.